server/src/test/java/org/elasticsearch/common/time/DateFormattersTests.java

```diff
@@ -695,6 +695,23 @@ public void testMinMillis() {
         assertThat(javaFormatted, equalTo("-292275055-05-16T16:47:04.192Z"));
     }
 
+    public void testMinNanos() {
+        String javaFormatted = DateFormatter.forPattern("strict_date_optional_time").formatNanos(Long.MIN_VALUE);
+        assertThat(javaFormatted, equalTo("1677-09-21T00:12:43.145Z"));
+
+        // It is deeply unclear to me why the last 6 digits of this aren't 775808, which are the last 6 digits of Long.MIN_VALUE
```

Comment:
It's because `Long.MIN_VALUE % 1_000_000` is a negative value, resulting in the nanos being subtracted from the seconds rather than added

Reply:
oh, yeah, that makes sense.  And seems correct.  I'll update the comment before merging.